### PR TITLE
SONARJAVA-1643 Reimplement SuppressWarningFilter

### PR DESCRIPTION
**Pull request by m-g-sonar**
-100 lines!

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      }
+    }
+    return false;
+  }
+
+  private static boolean isRuleKEy(String rule, RuleKey ruleKey) {
+    try {
+      // format of the rules requires a repository: "repo:key"
+      return ruleKey.equals(RuleKey.parse(rule));
+    } catch (IllegalArgumentException e) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: CRITICAL Either log or rethrow this exception. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

File: java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        Collection<Integer> excludedLines = excludedLineByRule.get(excludedRule);
+        if (excludedLines.contains(issue.line())) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private static boolean isRuleKEy(String rule, RuleKey ruleKey) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: typo

File: java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      if (isSuppressWarningsAnnotation(annotationTree)) {
+        startLine = ((JavaTree) annotationTree).getLine();
+        rules.addAll(getRules(annotationTree));
+        break;
+      }
+    }
+
+    if (startLine != -1) {
+      int endLine = LastSyntaxTokenFinder.lastSyntaxToken(tree).line();
+      Set<Integer> filteredlines = Sets.newHashSet(Ranges.closed(startLine, endLine).asSet(DiscreteDomains.integers()));
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: no need to call newHashSet if I understand correctly guava documentation.

File: java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  @Override
+  public Set<Class<? extends JavaCheck>> filteredRules() {
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public boolean accept(Issue issue) {
+    if (issueShouldNotBeReported(issue, excludedLinesByRule())) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: could be simplified.

File: java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      }
+    }
+    return false;
+  }
+
+  private static boolean isRuleKey(String rule, RuleKey ruleKey) {
+    try {
+      // format of the rules requires a repository: "repo:key"
+      return ruleKey.equals(RuleKey.parse(rule));
+    } catch (IllegalArgumentException e) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java]: CRITICAL Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Pull request merged**